<!-- TOC -->

- [1. 抛出异常相关关键字](#1-抛出异常相关关键字)
- [2. throw 关键字](#2-throw-关键字)
  - [2.1. 执行语法](#21-执行语法)
  - [2.2. 应用场景](#22-应用场景)
- [3. throws 关键字](#3-throws-关键字)

<!-- /TOC -->

## 1. 抛出异常相关关键字
- `throw` 关键字
- `throws` 关键字

## 2. throw 关键字

### 2.1. 执行语法
- 运用于方法内部, 抛出一个具体的异常对象. 然后终止方法, 类似 `return`.
  ```java
  throw new SomeException("Reason of Exception");
  ```

- 这个抛出的异常可以在 `catch` 中截取并获取其中的信息.

### 2.2. 应用场景
- 一般来说, 当一个方法出现异常的时候, 此时用 `return` 难以定义返回的内容,   
  因此可用 `throw` 返回一个异常对象, 在 `catch` 代码块中继续向上抛出异常.  
  所以必须先在 `try` 代码块中对该方法进行调用, 才能 `catch` 到异常.

- `return` 是返回一个值, `throw` 是返回一个错误, 返回给方法的调用者.

## 3. throws 关键字
